### DISCUSSION TITLE
Best way to markup old price?

### DISCUSSION THREAD

**Discussion by zaitovalisher** [schemaorg/schemaorg#3035] [Q&A]
Hi,
I’ve found numerous opinions on that (listed below). What is the best way to markup old price?

related schema:

https://schema.org/SalePrice

https://schema.org/price

https://schema.org/Product

people’s opinions:

https://stackoverflow.com/questions/37860722/show-discount-in-schema-org

https://stackoverflow.com/questions/59768918/schema-org-pricespecification-with-a-sale-price

Google:
https://developers.google.com/search/docs/data-types/product

**Comment by RichardWallis** [comment]
I would suggest that for each [old] price you add an *Offer,* to the
*Product* or *Service,* with the appropriate *price*, *validFrom*, and
*validThrough* values.

This, in addition to the current price Offer(s), could capture a history of
prices up to and including the current.

Not sure who would consume such data, but that would be a logical way to
approach this.

~Richard.

On 19 Jan 2022 at 14:03:05, zaitovalisher ***@***.***> wrote:

> Hi,
> I’ve found numerous opinions on that (listed below). What is the best way
> to markup old price?
>
> related schema:
>
> https://schema.org/SalePrice
>
> https://schema.org/price
>
> https://schema.org/Product
>
> people’s opinions:
>
> https://stackoverflow.com/questions/37860722/show-discount-in-schema-org
>
>
> https://stackoverflow.com/questions/59768918/schema-org-pricespecification-with-a-sale-price
>
> Google:
> https://developers.google.com/search/docs/data-types/product
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/schemaorg/schemaorg/discussions/3035>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADFS2TTSLLEYLFR65JCL6ZTUW3ABTANCNFSM5MKASVFQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Richard Wallis
Founder, Data Liberate
http://dataliberate.com
Linkedin: http://www.linkedin.com/in/richardwallis
Twitter: @dataliberate @rjw

**Reply by zaitovalisher** [reply]
> I would suggest that for each [old] price you add an *Offer,* to the *Product* or *Service,* with the appropriate *price*, *validFrom*, and *validThrough* values. This, in addition to the current price Offer(s), could capture a history of prices up to and including the current. Not sure who would consume such data, but that would be a logical way to approach this. ~Richard.
> […](#)

Thanks for your answer, Richard!

As for the goal, I want to markup every inch of the category page on my website. So I hope it will give me some benefits in terms of SEO

**Comment by thadguidry** [comment]
Actually, I think you could also use CompoundPriceSpecification and set the priceType to use the enumeration for SalePrice versus a ListPrice
